peak_finder/src/peak_finder_component.cpp

```diff
@@ -0,0 +1,218 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+#include <stsl_interfaces/action/park_at_peak.hpp>
+#include <stsl_interfaces/srv/sample_elevation.hpp>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_eigen/tf2_eigen.h>
+#include <Eigen/Dense>
+#include <nav2_msgs/action/navigate_to_pose.hpp>
+#include <memory>
+#include <string>
+#include <vector>
+#include "navigator.hpp"
+
+namespace peak_finder
+{
+class PeakFinderComponent : public rclcpp::Node
+{
+public:
+  using ParkAtPeak = stsl_interfaces::action::ParkAtPeak;
+  using ParkAtPeakGoalHandle = rclcpp_action::ServerGoalHandle<ParkAtPeak>;
+
+  explicit PeakFinderComponent(const rclcpp::NodeOptions & options)
+  : rclcpp::Node("peak_finder", options), tf_buffer_(get_clock()), tf_listener_(tf_buffer_),
+    navigator_(*this)
+  {
+    declare_parameter<double>("search_radius", 0.1);
+    declare_parameter<int>("sample_count", 8);
+
+    action_server_ = rclcpp_action::create_server<ParkAtPeak>(
+      this, "park_at_peak",
+      std::bind(
+        &PeakFinderComponent::handle_goal, this, std::placeholders::_1,
+        std::placeholders::_2),
+      std::bind(&PeakFinderComponent::handle_cancel, this, std::placeholders::_1),
+      std::bind(&PeakFinderComponent::handle_accepted, this, std::placeholders::_1));
+
+    elevation_client_ = create_client<stsl_interfaces::srv::SampleElevation>("/sample_elevation");
+  }
+
+private:
+  tf2_ros::Buffer tf_buffer_;
+  tf2_ros::TransformListener tf_listener_;
+  rclcpp_action::Server<ParkAtPeak>::SharedPtr action_server_;
+  rclcpp::Client<stsl_interfaces::srv::SampleElevation>::SharedPtr elevation_client_;
+  Navigator navigator_;
+
+  rclcpp_action::GoalResponse handle_goal(
+    const rclcpp_action::GoalUUID & /*uuid*/,
+    std::shared_ptr<const ParkAtPeak::Goal>/*goal*/)
+  {
+    return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+  }
+
+  rclcpp_action::CancelResponse
+  handle_cancel(const std::shared_ptr<ParkAtPeakGoalHandle> goal_handle)
```

Comment:
If this passes the linter ignore me, this just feel weird

Reply:
Yeah, this is the linter's doing. The line was too long. 
Breaking up the return type and function name is kinda common for shortening lines like this, so I'm not upset about it, but yeah it can be weird.